tests/runner/check_runner.py

```diff
@@ -68,20 +70,33 @@ def check_simple_run(self):
         mock_cluster = LocalhostCluster(num_nodes=1000)
         session_context = tests.ducktape_mock.session_context()
 
-        test_methods = [TestThingy.test_pi, TestThingy.test_ignore1, TestThingy.test_ignore2]
+        test_methods = [TestThingy.test_pi, TestThingy.test_ignore1, TestThingy.test_ignore2, TestThingy.test_failure]
         ctx_list = self._do_expand(test_file=TEST_THINGY_FILE, test_class=TestThingy, test_methods=test_methods,
                                    cluster=mock_cluster, session_context=session_context)
         runner = TestRunner(mock_cluster, session_context, Mock(), ctx_list)
 
         results = runner.run_all_tests()
-        assert len(results) == 3
-        assert results.num_failed == 0
+        assert len(results) == 4
+        assert results.num_failed == 1
         assert results.num_passed == 1
         assert results.num_ignored == 2
 
         result_with_data = [r for r in results if r.data is not None][0]
         assert result_with_data.data == {"data": 3.14159}
 
+    def check_runner_report_xunit(self):
+        """Check we can serialize results into a xunit xml format"""
+        mock_cluster = LocalhostCluster(num_nodes=1000)
+        session_context = tests.ducktape_mock.session_context()
+        test_methods = [TestThingy.test_pi, TestThingy.test_ignore1, TestThingy.test_ignore2, TestThingy.test_failure]
+        ctx_list = self._do_expand(test_file=TEST_THINGY_FILE, test_class=TestThingy, test_methods=test_methods,
+                                   cluster=mock_cluster, session_context=session_context)
+        runner = TestRunner(mock_cluster, session_context, Mock(), ctx_list)
+
+        results = runner.run_all_tests()
+        XUnitReporter(results).report()
+        assert os.path.exists(os.path.join(session_context.results_dir, "report.xml"))
```

Comment:
Would be good to also test that the result actually matches what you expect it to generate, not just that the file exists.

Reply:
Sure I'll add in some xpath asserts here...

Reply:
Thanks! You could do xpath; or load this doc back into python objects and make asserts that way; or prepare a "golden" output doc, load both the doc generated in the test and the golden one in python and just `==` the resulting dicts.

Reply:
I opted for a more unit-driven approach with xpath (ie: unit test the number of testcase types in the XML doc).